### DISCUSSION TITLE
TASKDATA.xml file acquisition

### DISCUSSION THREAD

**Discussion by thiagozf1** [Open-Agriculture/AgIsoStack-plus-plus#286] [Q&A] [answered]
Hi guys, it's me again... Is there any way to request the TASKDATA.xml file on the bus? I didn't find any function at the TaskController functions (objects, client, etc) that could do this. I don't if this is possible to do, but it would be really usefull if it was.

**Comment by ad3154** [comment, marked as answer]
So, the TASKDATA.xml file is maintained by the task controller server, and is comprised of the DDOPs it has received from various TC clients once they are converted back to XML from their binary form. As far as I know, there's no standard way to request this from the task controller via CAN, other than to be the task controller yourself (which is a feature I hope to add at some point - the hardest or at least most arcane part of that was done in #276 ) or sniff all the DDOP uploads from other devices and re-construct the XML yourself, which doesn't sound super fun.

**Reply by thiagozf1** [reply]
Could the TASKDATA.xml be saved in a FileServer and than requested by a client?

**Reply by ad3154** [reply]
I suppose that could work. File server is a strange beast. Most implementations I've seen generally force you to read and write within a special directory corresponding to your manufacturer code on each volume. So for example, if your manufacturer code is 64, it expects you to work within a directory called "MCMC0064". But in theory, as far as I know, there's nothing saying you couldn't put it in some non-manufacturer directory and allow all users to read it, or put it in your MCMC folder and allow your specific clients to read it.

I'm hoping to have all parts, server and client, of all the ISOBUS protocols in here at some point including the 11783-13 server-side 😊 it's all about how much time it takes of course
